From f835e03e88ba3cb8de941c7cd0862c57afe22da5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Elan=20Ruusam=C3=A4e?= <glen@delfi.ee>
Date: Sun, 24 Jun 2012 12:23:30 +0300
Subject: [PATCH] introduce DOKU_DATA and DOKU_CACHE variables for packaging

add new constants, that makes packaging (rpm, deb) dokuwiki better fit
to [FHS standards](http://www.pathname.com/fhs/pub/fhs-2.3.html):
- DOKU_DATA - /var/lib/dokuwiki/
- DOKU_CACHE - /var/cache/dokuwiki/
---
 inc/init.php |    8 +++++++-
 install.php  |   26 ++++++++++++++------------
 2 files changed, 21 insertions(+), 13 deletions(-)

diff --git a/inc/init.php b/inc/init.php
index 403fbe4..29d96b2 100644
--- a/inc/init.php
+++ b/inc/init.php
@@ -23,7 +23,13 @@ if(!defined('DOKU_INC')) define('DOKU_INC',fullpath(dirname(__FILE__).'/../').'/
 if(!defined('DOKU_PLUGIN'))  define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');
 
 // define config path (packagers may want to change this to /etc/dokuwiki/)
-if(!defined('DOKU_CONF')) define('DOKU_CONF',DOKU_INC.'conf/');
+if(!defined('DOKU_CONF')) define('DOKU_CONF','/etc/webapps/dokuwiki/');
+
+// define path where variable data is saved (packagers may want to change this to /var/lib/dokuwiki/)
+if(!defined('DOKU_DATA')) define('DOKU_DATA','/var/lib/dokuwiki/');
+
+// define path where cache data saved (packagers may want to change this to /var/cache/dokuwiki/)
+if(!defined('DOKU_CACHE')) define('DOKU_CACHE','/var/cache/dokuwiki/');
 
 // check for error reporting override or set error reporting to sane values
 if (!defined('DOKU_E_LEVEL') && @file_exists(DOKU_CONF.'report_e_all')) {
diff --git a/install.php b/install.php
index 80e9ad4..fe72513 100644
--- a/install.php
+++ b/install.php
@@ -7,7 +7,9 @@
 
 if(!defined('DOKU_INC')) define('DOKU_INC',dirname(__FILE__).'/');
 if(!defined('DOKU_CONF')) define('DOKU_CONF',DOKU_INC.'conf/');
-if(!defined('DOKU_LOCAL')) define('DOKU_LOCAL',DOKU_INC.'conf/');
+if(!defined('DOKU_LOCAL')) define('DOKU_LOCAL',DOKU_CONF);
+if(!defined('DOKU_DATA')) define('DOKU_DATA',DOKU_INC.'data');
+if(!defined('DOKU_CACHE')) define('DOKU_CACHE',DOKU_DATA.'cache');
 
 require_once(DOKU_INC.'inc/PassHash.class.php');
 
@@ -428,17 +430,17 @@ function check_permissions(){
 
     $dirs = array(
         'conf'        => DOKU_LOCAL,
-        'data'        => DOKU_INC.'data',
-        'pages'       => DOKU_INC.'data/pages',
-        'attic'       => DOKU_INC.'data/attic',
-        'media'       => DOKU_INC.'data/media',
-        'media_attic' => DOKU_INC.'data/media_attic',
-        'media_meta'  => DOKU_INC.'data/media_meta',
-        'meta'        => DOKU_INC.'data/meta',
-        'cache'       => DOKU_INC.'data/cache',
-        'locks'       => DOKU_INC.'data/locks',
-        'index'       => DOKU_INC.'data/index',
-        'tmp'         => DOKU_INC.'data/tmp'
+        'data'        => DOKU_DATA,
+        'pages'       => DOKU_DATA.'pages',
+        'attic'       => DOKU_DATA.'attic',
+        'media'       => DOKU_DATA.'media',
+        'media_attic' => DOKU_DATA.'media_attic',
+        'media_meta'  => DOKU_DATA.'media_meta',
+        'meta'        => DOKU_DATA.'meta',
+        'cache'       => DOKU_CACHE,
+        'locks'       => DOKU_DATA.'locks',
+        'index'       => DOKU_DATA.'index',
+        'tmp'         => DOKU_DATA.'tmp'
     );
 
     $ok = true;
-- 
1.7.7.4
